test/e2e/clusterclass_rollout.go

```diff
@@ -1351,12 +1367,16 @@ func modifyMachinePoolViaClusterAndWait(ctx context.Context, input modifyMachine
 	}
 }
 
-func expectMapsToBeEquivalent(g Gomega, m1, m2 map[string]string) {
+func expectMapsToBeEquivalent(g Gomega, m1, m2 map[string]string, ignoreKeys ...string) {
 	if m1 == nil {
 		m1 = map[string]string{}
 	}
 	if m2 == nil {
 		m2 = map[string]string{}
 	}
+	for _, key := range ignoreKeys {
```

Comment:
Can we please clone the maps before we modify them here?

(fine as a follow-up PR)